server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -249,34 +313,50 @@ export class TenantManager {
 
         const encryptedTenantKey1 = tenantDocument.key;
         const tenantKey1 = this.secretManager.decryptSecret(encryptedTenantKey1);
+        let tenantKey2;
+
         if (tenantKey1 == null) {
             winston.error("Tenant key1 decryption failed.");
-            Lumberjack.error("Tenant key1 decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+            Lumberjack.error("Tenant key1 decryption failed.", lumberProperties);
             throw new NetworkError(500, "Tenant key1 decryption failed.");
         }
 
         const encryptedTenantKey2 = tenantDocument.secondaryKey;
-        if (!encryptedTenantKey2) {
+        if (encryptedTenantKey2) {
+            tenantKey2 = this.secretManager.decryptSecret(encryptedTenantKey2);
+            if (tenantKey2 == null) {
+                winston.error("Tenant key2 decryption failed");
+                Lumberjack.error("Tenant key2 decryption failed.", lumberProperties);
+                throw new NetworkError(500, "Tenant key2 decryption failed.");
+            }
+        } else {
             winston.info("Tenant key2 doesn't exist.");
-            Lumberjack.info("Tenant key2 doesn't exist.", { [BaseTelemetryProperties.tenantId]: tenantId });
-            return {
-                key1: tenantKey1,
-                key2: "",
-            };
+            Lumberjack.info("Tenant key2 doesn't exist.", lumberProperties);
         }
 
-        const tenantKey2 = this.secretManager.decryptSecret(encryptedTenantKey2);
-        if (tenantKey2 == null) {
-            winston.error("Tenant key2 decryption failed.");
-            Lumberjack.error("Tenant key2 decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
-            throw new NetworkError(500, "Tenant key2 decryption failed.");
-        }
+        const cacheKeys = {
+            key1: encryptedTenantKey1,
+            key2: encryptedTenantKey2,
+        };
+
+        await this.setCacheKey(tenantId, cacheKeys, fetchTenantKeyMetric);
 
         return {
             key1: tenantKey1,
             key2: tenantKey2,
         };
+    } catch (error) {
+        uncaughtException = error;
+        throw error;
+    } finally {
+        fetchTenantKeyMetric.setProperties({ [BaseTelemetryProperties.tenantId]: tenantId, includeDisabledTenant, retrievedFromCache });
+        if (!uncaughtException) {
+            fetchTenantKeyMetric.success(`Successfully retrieved tenant keys.`);
+        } else {
+            fetchTenantKeyMetric.error(`Error trying to retrieve tenant keys.`);
```

Comment:
I like this new setup! Definitely cleaner than the draft code I had prepared in our conversation 😄 Just one more detail: let's update the `.error()` call to pass the exception object to lumberjack, so we have more info on what error happened. Like this:
```suggestion
            fetchTenantKeyMetric.error(`Error trying to retrieve tenant keys.`, uncaughtException);
```